### DISCUSSION TITLE
Filter visible collections items based on M2M relation

### DISCUSSION THREAD

**Discussion by henninghorn** [directus/directus#4967] [Archive]
Hi!

I am really excited about Directus and how I may be able to use it future projects, and I hope to be able contribute when I have learned the inner workings of the code base.

Currently I am trying to figure how to filter the data displayed to a user.
The docs only have an example of filtering a O2M relation:

<img width="777" alt="billede" src="https://user-images.githubusercontent.com/742607/114294315-038cc700-9a9e-11eb-9f6b-ada37d6aa623.png">

How do you that on a M2M relation?

I have a `organizations` collection with a M2M relation to the users, which also has a `role`-field to determine the user's relation to the organisation, e.g. "employee", "director" etc. Those directus users has a directus role where I customize the permissions `organizations` and where I try to add the filter.

When a regular user log in and navigates to the `organization` collection, I want to only display the organizations the user has a relation to (through the M2M relation).
How do I set up a filter to do that?

**Comment by henninghorn** [comment]
I believe I found the solution to be:

Setting the following on the role's "view"-permission on the organization collection
```json
{
  "members": {
    "user_id": {
      "_contains": "$CURRENT_USER"
    }
  }
}
```

Please correct me if this should be achieved in another way.

**Reply by rijkvanzanten** [reply]
This should do the trick! `_eq` and `_contains` should be both have the desired effect in this case
